### PR TITLE
Expose API to set log level in embedder

### PR DESCRIPTION
**Pull request by eloparco**

### CODE REVIEW

**Review by wenyongh — APPROVED** [review]
LGTM

File: core/iwasm/include/wasm_export.h [hunk 1]
```diff
@@ -234,6 +242,14 @@ wasm_runtime_init(void);
 WASM_RUNTIME_API_EXTERN bool
 wasm_runtime_full_init(RuntimeInitArgs *init_args);
 
+/**
+ * Set the log level. To be called after the runtime is created.
```
Comment (wenyongh) [inline core/iwasm/include/wasm_export.h:246]: Had better change `created` to `initialized`?

File: core/iwasm/include/wasm_export.h [hunk 2]
```diff
@@ -212,6 +212,14 @@ typedef struct wasm_val_t {
 } wasm_val_t;
 #endif
 
+typedef enum {
+    BH_LOG_LEVEL_FATAL = 0,
+    BH_LOG_LEVEL_ERROR = 1,
+    BH_LOG_LEVEL_WARNING = 2,
+    BH_LOG_LEVEL_DEBUG = 3,
+    BH_LOG_LEVEL_VERBOSE = 4
```
Comment (yamt) [inline core/iwasm/include/wasm_export.h:220]: if these are for wasm_runtime_set_log_level, it seems a bit odd to use BH_ prefix. how do you think?

Comment (wenyongh) [reply core/iwasm/include/wasm_export.h:220]: How about changing to `WASM_LOG_LEVEL_*`? Or is there other suggestion?

Comment (yamt) [reply core/iwasm/include/wasm_export.h:220]: > How about changing to `WASM_LOG_LEVEL_*`?

it sounds better to me.

Comment (wenyongh) [reply core/iwasm/include/wasm_export.h:220]: Good to me too. @eloparco, could you help update the code? Thanks.

File: core/iwasm/include/wasm_export.h [hunk 3]
```diff
@@ -212,6 +212,14 @@ typedef struct wasm_val_t {
 } wasm_val_t;
 #endif
 
+typedef enum {
+    WASM_LOG_LEVEL_FATAL = 0,
+    WASM_LOG_LEVEL_ERROR = 1,
+    WASM_LOG_LEVEL_WARNING = 2,
+    WASM_LOG_LEVEL_DEBUG = 3,
+    WASM_LOG_LEVEL_VERBOSE = 4
```
Comment (wenyongh) [inline core/iwasm/include/wasm_export.h:220]: Should also change all "BH_LOG_" related strings in the source code to "WASM_LOG_*".

Comment (eloparco) [reply core/iwasm/include/wasm_export.h:220]: Yes, I was about to comment, do we want to do that or keep the internal name as `BH_LOG_LEVEL_*` and use `WASM_LOG_LEVEL_*` only when exposing?

Comment (eloparco) [reply core/iwasm/include/wasm_export.h:220]: By re-introducing
```c
typedef enum {
    BH_LOG_LEVEL_FATAL = 0,
    BH_LOG_LEVEL_ERROR = 1,
    BH_LOG_LEVEL_WARNING = 2,
    BH_LOG_LEVEL_DEBUG = 3,
    BH_LOG_LEVEL_VERBOSE = 4
} log_level_t;
```
in `bh_log.h`.

Probably better to only keep `WASM_LOG_LEVEL_ *` and change all occurrences, I'll do that.

Comment (wenyongh) [reply core/iwasm/include/wasm_export.h:220]: Seems better to keep "BH_LOG_*" in bh_log.h and convert "WASM_LOG_*" to "BH_LOG_*" in wasm_runtime_set_log_level?

**Comment by eloparco** [comment]
```bash
[build_iwasm ($CLASSIC_INTERP_BUILD_OPTIONS, -DWAMR_BUILD_LIB_PTHREAD=1, macos-latest, darwin)](https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/7220553955/job/19673804322)
GitHub Actions has encountered an internal error when running your job.
```
I'm amending the commit to restart the job.

**Pull request merged**